src/fontra/views/editor/panel-transformation.js

```diff
@@ -450,11 +584,334 @@ export default class TransformationPanel extends Panel {
+      const individualSelection = pointIndices.map(
+        (pointIndex) => `point/${pointIndex}`
+      );
+      movableObjects.push(new MovableContour(contourIndex, individualSelection));
+    }
+    for (const componentIndex of components) {
+      const individualSelection = [`component/${componentIndex}`];
+      movableObjects.push(new MovableComponent(componentIndex, individualSelection));
+    }
+    return movableObjects;
+  }
+
+  async moveObjects(moveDescriptor) {
+    const staticGlyphControllers = await this._getStaticGlyphControllers();
+
+    await this.sceneController.editGlyph((sendIncrementalChange, glyph) => {
+      const editLayerGlyphs = this.sceneController.getEditingLayerFromGlyphLayers(
+        glyph.layers
+      );
+
+      const editChanges = [];
+      const rollbackChanges = [];
+      for (const [layerName, layerGlyph] of Object.entries(editLayerGlyphs)) {
+        const changePath = ["layers", layerName, "glyph"];
+        const controller = staticGlyphControllers[layerName];
+
+        const movableObjects = this._collectMovableObjects(controller);
+        if (movableObjects.length <= 1) {
+          continue;
+        }
+
+        const boundingBoxes = movableObjects.map((obj) =>
+          obj.computeBounds(controller)
+        );
+        const deltas = moveDescriptor.computeDeltasFromBoundingBoxes(boundingBoxes);
+        for (const [delta, movableObject] of zip(deltas, movableObjects)) {
+          const [editChange, rollbackChange] = movableObject.getChangesForDelta(
+            delta,
+            layerGlyph,
+            this.sceneController
+          );
+          applyChange(layerGlyph, editChange);
+          editChanges.push(consolidateChanges(editChange, changePath));
+          rollbackChanges.push(consolidateChanges(rollbackChange, changePath));
+        }
+      }
+
+      let changes = ChangeCollector.fromChanges(
+        consolidateChanges(editChanges),
+        consolidateChanges(rollbackChanges)
+      );
+
+      return {
+        changes: changes,
+        undoLabel: moveDescriptor.undoLabel,
+        broadcast: true,
+      };
+    });
+  }
+
   async toggle(on, focus) {
     if (on) {
       this.update();
     }
   }
 }
 
+// Define MovableObject classes
+class MovableBaseObject {
+  constructor(selection) {
+    this.selection = selection;
+  }
+
+  computeBounds(staticGlyphController) {
+    return staticGlyphController.getSelectionBounds(this.selection);
+  }
+
+  getChangesForDelta(delta, layerGlyph, sceneController) {
+    const behaviorFactory = new EditBehaviorFactory(
+      layerGlyph,
+      this.selection,
+      sceneController.experimentalFeatures.scalingEditBehavior
+    );
+
+    const t = new Transform().translate(delta.x, delta.y);
+    const pointTransformFunction = t.transformPointObject.bind(t);
+    const editBehavior = behaviorFactory.getBehavior("default");
+    const editChange = editBehavior.makeChangeForTransformFunc(pointTransformFunction);
```

Comment:
You can just use `makeChangeForDelta` and save the trouble with the Transform object.

Reply:
Of cause. I forgot about it. Thanks for the hint. Done.